container/container.go

```diff
@@ -389,11 +390,18 @@ func AttachStreams(ctx context.Context, streamConfig *runconfig.StreamConfig, op
 
 		var err error
 		if tty {
-			_, err = copyEscapable(cStdin, stdin, keys)
+			var written int64
+			written, err = copyEscapable(cStdin, stdin, keys)
+			if written == -1 { // detached
```

Comment:
I don't think we should be using -1 as meaning "detached", can we return an extra boolean instead?

Most system developers are used of reading -1 as "an error occurred", and although errors are reported via a special error type in golang, I think it may still be good idea not to infringe on this convention.

wdyt?